lib-es5/uploader.js

```diff
@@ -485,26 +487,27 @@ function call_api(action, callback, options, get_params) {
         result = parseResult(buffer, res);
         if (result.error) {
           result.error.http_code = res.statusCode;
-          deferred.reject(result.error);
+          handleUploadAPIError(callback, result, deferred, result.error);
         } else {
           cacheResults(result, options);
-          deferred.resolve(result);
+          if (callback) {
+            callback(result);
+          } else {
+            deferred.resolve(result);
+          }
```

Comment:
@strausr , why not calling the handleError?

Reply:
Resolving. We can make a resolving handler?